sdk/tables/azure-data-tables/azure/data/tables/aio/_base_client_async.py

```diff
@@ -153,8 +153,11 @@ async def _batch_send(self, table_name: str, *reqs: HttpRequest, **kwargs) -> Li
 
         parts_iter = response.parts()
         parts = []
-        async for p in parts_iter:
-            parts.append(p)
+        try:
+            async for p in parts_iter:
+                parts.append(p)
+        except IndexError:
```

Comment:
Is it really this line that's causing the IndexError? Where is this IndexError actually triggered from? I would have expected in line 154...

Reply:
Yeah, I also thought it failed in getting the result. But it actually failed in iterating.